### PR TITLE
DEV: Make `capabilities` into a service

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

**Review by davidtaylorhq — APPROVED** [review]

**Comment by davidtaylorhq** [comment]
Oh, I think we should add a deprecation for `capabilities:main` in https://github.com/discourse/discourse/blob/91b991320c267aad6ba88dde26f9fe5729508689/app/assets/javascripts/discourse-common/addon/resolver.js#L12-L12

**Comment by jjaffeux** [comment]
@CvX I think this one would be worth finishing, it's something I have been missing again this week in glimmer components cc @davidtaylorhq

**Comment by davidtaylorhq** [comment]
Yup, would be great to get this merged

> it's something I have been missing again this week in glimmer components

You may already be doing this, but a workaround to unblock using `capabilities` in a glimmer component can be:

```javascript
get capabilities(){
  return getOwner(this).lookup("capabilities:main");
}
```

**Comment by jjaffeux** [comment]
> Yup, would be great to get this merged
> 
> > it's something I have been missing again this week in glimmer components
> 
> You may already be doing this, but a workaround to unblock using `capabilities` in a glimmer component can be:
> 
> ```js
> get capabilities(){
>   return getOwner(this).lookup("capabilities:main");
> }
> ```

yep thats what I did indeed 👍

File: app/assets/javascripts/discourse/app/services/capabilities.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import Service from "@ember/service";
+
+const APPLE_NAVIGATOR_PLATFORMS = /iPhone|iPod|iPad|Macintosh|MacIntel/;
+const APPLE_USER_AGENT_DATA_PLATFORM = /macOS/;
+
+// Lets us know about browser's capabilities
+export default Service.extend({
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/services/capabilities.js:7]: Can we make it use native class syntax while we're here?

**Pull request merged**